
<a href='https://www.darshan.ac.in/'> <img src='https://www.darshan.ac.in/Content/media/DU_Logo.svg' width="250" height="300"/></a>
<pre>
<center><b><h1>Python Programming - 2301CS404</b></center>
<center><b><h1>Lab - 4</b></center>    
<pre>

# String

### 01) WAP to check whether the given string is palindrome or not.

In [22]:
str=input("enter your string")

rev_str = str[::-1]

if(rev_str==str):
    print("string is palindrome")
    
else:
    print("string is palindrome")

enter your string1234
string is palindrome


### 02) WAP to reverse the words in the given string.

In [ ]:
str=input("enter your string")

rev_str = str[::-1]

print(rev_string)

### 03) WAP to remove ith character from given string.

In [23]:
str=input("enter your string")
i=int(input("enter your index"))
ans=str[0:i]+str[i+1:]
print(ans)

enter your string1234
enter your index1
134


### 04) WAP to find length of string without using len function.

In [29]:
str=input("enter your string")
ans=0
for i in str:
    ans+=1
print(ans)

enter your string1234
4


### 05) WAP to print even length word in string.

In [31]:
str=input("enter your string")
s=str.split(" ")
for i in s:
    if(len(i)%2==0):
        print(i)

enter your string1234 123
1234


### 06) WAP to count numbers of vowels in given string.

In [37]:
str=input("enter your string")
s=str.split(" ")
count=0
for x in str:
    if(x=="a" or x=="A" or x=="e" or x=="e" or x=="i" or x=="I" or x=="o" or x=="O" or x=="u" or x=="U"):
        count+=1
print(count)

enter your stringabe
2


### 07) WAP to capitalize the first and last character of each word in a string.

In [2]:
str=input("enter your string")
words = str.split()
result = []

for word in words:
    if len(word) > 1:
        result.append(word[0].upper() + word[1:-1] + word[-1].upper())
    else:
        result.append(word.upper())

print(" ".join(result))


enter your stringraj vekariya
RaJ VekariyA


### 08) WAP to convert given array to string.

In [56]:
g=['e','b','c']
z=""
for j in g:
    z=z+j
print(z)  

ebc


### 09) Check if the password and confirm password is same or not. 
### In case of only case's mistake, show the error message.

In [4]:
password=input("enter your string")
con_password=input("enter your string")

if(password==con_password):
    print("password is same")
elif(password.lower()==con_password.lower()):
    print('case error')
else:
    print("pass is not same")

enter your stringraj patel
enter your stringRaj Patel
case error


### 10) : Display credit card number. 
### card no. : 1234 5678 9012 3456
### display as  : **** **** **** 3456

In [8]:
string=input("enter your credit card number")
x=0
for x in range(0,19):
    if(x<15):
        print("*",end=" ")
    else:
        print(string[x],end=" ")
            
    

enter your credit card number1234 5678 9012 3456
* * * * * * * * * * * * * * * 3 4 5 6 

### 11) : Checking if the two strings are Anagram or not.
### s1 = decimal and s2 = medical are Anagram

In [2]:
s1 =input("enter your credit card number")
s2 =input("enter your credit card number")
if sorted(s1) == sorted(s2):
    print("Anagram")
else:
    print("Not anagram")


KeyboardInterrupt: Interrupted by user

### 12) : Rearrange the given string. First lowercase then uppercase alphabets.
### input : EHlsarwiwhtwMV
### output : lsarwiwhtwEHMV

In [ ]:
input_str =input("enter your credit card number")
lowercase = ""
uppercase = ""

for char in input_str:
    if char.islower():
        lowercase += char
    else:
        uppercase += char

output_str = lowercase + uppercase
print(output_str)
